### DISCUSSION TITLE
How to search for upercase letters with outline-regexp?

### DISCUSSION THREAD

**Discussion by AnabasisXu** [minad/consult#962] [Q&A] [answered]
I tried to search for `// ABC` but not `// abc` in the following text:

```python
// ABC
// abc
```
So I set `(setq outline-regexp "// [A-Z]")`. It matches both lines rather than just the line with uppercase letters.

**Comment by minad** [comment]
`outline-regexp` is case insensitive by default. One can theoretically change this by globally setting `case-fold-search` to nil, but I don't recommend this, since this will have unexpected effects for many packages.

**Reply by AnabasisXu** [reply]
Thank you @minad! There is currently no `consult` package setting option to make `outline-regexp` case sensitive right?

**Reply by oantolin** [reply, marked as answer]
You can define your own command like this:

```elisp
(defun case-sensitive-consult-outline (&optional level)
  (interactive
   (list (and current-prefix-arg (prefix-numeric-value current-prefix-arg))))
  (let ((case-fold-search nil))
    (consult-outline level)))
```
